### PR TITLE
auth: implemented login interface

### PR DESCRIPTION
**Pull request by doflamingo721**
1. login api consumption,
2. storing token in SharedPreferences

### CODE REVIEW

File: lib/src/api/login_api_service.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'dart:io';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class APIService {
+  Future<Map> login(String userId, String password) async {
+    dynamic postBody = <String, String>{"userId": userId, "password": password};
+    String url = "http://52.136.112.220:57842/api/auth/login";
```
Comment (drigio) [inline lib/src/api/login_api_service.dart:8]: Create a constant for `server address` url and then append it with your route. For eg here add constant `serverURL` to `libs/constants`. Import that constant here. then create a new constant `loginRoute` in this file, which has `serverURL+"/api/auth/login"`.

File: lib/src/api/login_api_service.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    var headers = {'Content-Type': 'application/json'};
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```
Comment (drigio) [inline lib/src/api/login_api_service.dart:10]: Save the `access_token` into `FlutterSecureStorage`. And save `is_logged_in` = true in `SharedPrefs`.

File: lib/src/model/login_access.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// class LoginResponseModel {
```
Comment (drigio) [inline lib/src/model/login_access.dart]: Dont git stage this file if you arent supposed to upload it to the remote repo.

File: lib/src/pages/profile_page.dart [hunk 4]
```diff
@@ -0,0 +1,465 @@
+import 'package:flutter/material.dart';
+import 'package:final_app/themes.dart';
+
+// class ProfilePage extends StatefulWidget {
```
Comment (drigio) [inline lib/src/pages/profile_page.dart]: Remove unnecessary comments and again push the code.

File: lib/themes.dart [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+
+const String kAppName = "Grievance Portal";
```
Comment (drigio) [inline lib/themes.dart:3]: Why is app name `string` constant stored in `themes.dart` file. Use `constants.dart` for the same.

**Review by drigio — CHANGES_REQUESTED** [review]

File: lib/src/api/login_api_service.dart (hunk 1 above)
Comment (doflamingo721) [reply lib/src/api/login_api_service.dart:8]: Yes, I will do the needful in the next PR.

**Pull request merged**